lldp_8023.c

```diff
@@ -463,6 +464,47 @@ void ieee8023_ifup(char *ifname, struct lldp_agent *agent)
 	return;
 }
 
+/*
+ * ieee8023_rchange: process received IEEE 802.3 TLV LLDPDU
+ *
+ * TLV not consumed on error
+ */
+int ieee8023_rchange(struct port *port, struct lldp_agent *agent,
+		     struct unpacked_tlv *tlv)
+{
+	struct ieee8023_data *bd;
+	u8 subtype;
+	u8 *oui;
+
+	if (agent->type != NEAREST_BRIDGE)
+		return SUBTYPE_INVALID;
+
+	bd = ieee8023_data(port->ifname, agent->type);
+	if (!bd)
+		return SUBTYPE_INVALID;
+
+	if (tlv->type != TYPE_127)
+		return SUBTYPE_INVALID;
+
+	if (tlv->length < OUI_SUB_SIZE)
+		return TLV_ERR;
+
+	oui = tlv->info;
+	if (ntoh24(oui) != OUI_IEEE_8023)
+		return SUBTYPE_INVALID;
+
+	subtype = *(tlv->info + OUI_SIZE);
+	switch (subtype) {
+	case LLDP_8023_MACPHY_CONFIG_STATUS:
+	case LLDP_8023_POWER_VIA_MDI:
+	case LLDP_8023_LINK_AGGREGATION:
+	case LLDP_8023_MAXIMUM_FRAME_SIZE:
+		return TLV_OK;
```

Comment:
If I understand correctly, we need to keep a reference to the TLV here and release it properly, *or* we need to introduce a new return code (something like TLV_OK_RELEASE) that will tell the rxProcessFrame to delete the TLV object.  Otherwise, I think there could be a memory leak.  Can you confirm?